### ISSUE TITLE
Parametrical templates

### ISSUE THREAD

**Issue by uliska** [frescobaldi/frescobaldi#1200]
"New from template" could easily be made more powerful by enabling (string) template replacements.

Suggestion:
If the template contains instances of `<<<key>>>` string templates a pop-up dialog appears providing GUI elements to insert values. The string templates will then replaced with the user-provided values.

Typical use cases may be titles, but also Time or key signatures and instrument names. But of course people will find other ways. to use that feature.

**Comment by maribote** [comment]
I'm not sure if this wish is somehow my own wish as well. Would this feature allow for entering a random number of "empty" measures for all instruments in the wizard? I found that having a score with, say, 50 measures with rests for all voices
R1
R1
R1
...
is a great way to start composing the way you would do in other software. Each measure can be clicked in the PDF viewer and the corresponding rest can be replaced with music in that particular line without having to count measures evry time, while allowing the user to change his mind later and "fill" other measures with music. For the parts those rests will have to be compressed in a later stage using the respective multipliers, but I think it's faster to do that manually than using R*50 from the beginning, which is already a more definitive way of writing measure rests (this last step could also be automatized with a "merge measure rests" function, but that's a different wish!).

Regards,
Martín Rincón Botero.
www.martinrinconbotero.com

**Comment by uliska** [comment]
I'm not sure this is naturally the case. Of course you could create your 50-rests-string and merge this into a `<<<rests>>>` string template that is present in all voices.However, my feeling is that your use case (which I find very intersting, BTW) would be better handled with something else, e.g. a Python snippet or an extension.

**Comment by maribote** [comment]
You're right. I guess I'll open a new thread. My wish is for the Score Wizard, but I suppose it could be implemented in many ways.

**Comment by uliska** [comment]
It might be an idea to have a checkbox and text field in the Acore Wizard that would add some initial content into every generated voice.

**Comment by maribote** [comment]
That's maybe more powerful than the empty measures idea. I like it! 

www.martinrinconbotero.com

**Comment by maribote** [comment]
"It might be an idea to have a checkbox and text field in the Acore Wizard that would add some initial content into every generated voice".

I imagine that, if this is implemented in the Score Wizard, it could be also implemented eventually outside of it, with a function called Add Code to All Instruments or similar, that adds content to all instruments at once. It could have the options "Add code to the beginning" and "Add code to the end".

www.martinrinconbotero.com
